packages/react/src/ActionList/Item.tsx

```diff
@@ -129,7 +129,7 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
     const itemRole = role || inferredItemRole
 
     if (slots.trailingAction) {
-      invariant(!container, `ActionList.TrailingAction can not be used within a ${container}.`)
+      invariant(container !== 'ActionMenu', `ActionList.TrailingAction can not be used within a ${container}.`)
```

Comment:
@tylerjdev, I could not get the TrailingAction to render in `NavList` without updating the logic in this file. Do you mind double checking this? 

Related prior commit: https://github.com/primer/react/pull/4634/commits/8802b79b5fb62a3d552255a8c6561578746ea894

Reply:
We'll need to ensure that it's neither an "ActionMenu" or a "SelectPanel". It might be best to put this in a const above. I think this would work:

```javascript
const menuContext = container === 'ActionMenu' || container === 'SelectPanel'`
invariant(!menuContext, `ActionList.TrailingAction can not be used within a ${container}.`)
```

Reply:
```suggestion
      const menuContext = container === 'ActionMenu' || container === 'SelectPanel'`
      invariant(!menuContext, `ActionList.TrailingAction can not be used within a ${container}.`)
```